x/concentrated-liquidity/position.go

```diff
@@ -561,21 +566,46 @@ func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64,
 	return sdk.BigEndianToUint64(value), nil
 }
 
-// SetPositionIdToLock sets the positionId to lock mapping in state.
+// GetPositionIdFromLockId returns the lockId to position mapping in state.
+func (k Keeper) GetPositionIdFromLockId(ctx sdk.Context, underlyingLockId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the lock ID to position ID mapping.
+	key := types.KeyLockIdForPositionId(underlyingLockId)
+	value := store.Get(key)
+	if value == nil {
+		return 0, types.LockIdToPositionIdNotFoundError{LockId: underlyingLockId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
+	// Get the position ID to key mapping and set.
 	key := types.KeyPositionIdForLock(positionId)
 	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+
+	// Get the lock ID to key mapping and set.
+	key = types.KeyLockIdForPositionId(underlyingLockId)
+	store.Set(key, sdk.Uint64ToBigEndian(positionId))
```

Comment:
Checking that there is a test validating that this index is updated?

Reply:
Added the reverse check (lock to position) here https://github.com/osmosis-labs/osmosis/pull/4885/commits/a2f6c7a3f231c74196226ec7e9cfde10868b55b5